libraries/exoplayer_rtsp/src/main/java/androidx/media3/exoplayer/rtsp/reader/RtpMp4aPayloadReader.java

```diff
@@ -0,0 +1,157 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package androidx.media3.exoplayer.rtsp.reader;
+
+import static androidx.media3.common.util.Assertions.checkState;
+import static androidx.media3.common.util.Assertions.checkStateNotNull;
+import static androidx.media3.common.util.Assertions.checkArgument;
+import static androidx.media3.common.util.Util.castNonNull;
+
+import androidx.annotation.Nullable;
+import androidx.media3.common.C;
+import androidx.media3.common.ParserException;
+import androidx.media3.common.util.ParsableBitArray;
+import androidx.media3.common.util.ParsableByteArray;
+import androidx.media3.common.util.UnstableApi;
+import androidx.media3.common.util.Util;
+import androidx.media3.exoplayer.rtsp.RtpPayloadFormat;
+import androidx.media3.extractor.ExtractorOutput;
+import androidx.media3.extractor.TrackOutput;
+
+import com.google.common.collect.ImmutableMap;
+import org.checkerframework.checker.nullness.qual.MonotonicNonNull;
+
+/**
+ * Parses an MP4A-LATM byte stream carried on RTP packets, and extracts MP4A-LATM Access Units.
+ * Refer to RFC3016 for more details.
+ */
+@UnstableApi
+/* package */ final class RtpMp4aPayloadReader  implements RtpPayloadReader {
+  private static final String TAG = "RtpMp4aLatmReader";
+
+  private static final String PARAMETER_MP4A_CONFIG = "config";
+
+  private final RtpPayloadFormat payloadFormat;
+  private @MonotonicNonNull TrackOutput trackOutput;
+  private long firstReceivedTimestamp;
+  /** The combined size of a sample that is fragmented into multiple subFrames. */
+  private int fragmentedSampleSizeBytes;
+  private long startTimeOffsetUs;
+
+  /** Creates an instance. */
+  public RtpMp4aPayloadReader(RtpPayloadFormat payloadFormat) {
+    this.payloadFormat = payloadFormat;
+    firstReceivedTimestamp = C.TIME_UNSET;
+    fragmentedSampleSizeBytes = 0;
+    // The start time offset must be 0 until the first seek.
+    startTimeOffsetUs = 0;
+  }
+
+  @Override
+  public void createTracks(ExtractorOutput extractorOutput, int trackId) {
+    trackOutput = extractorOutput.track(trackId, C.TRACK_TYPE_VIDEO);
+    castNonNull(trackOutput).format(payloadFormat.format);
+  }
+
+  @Override
+  public void onReceivingFirstPacket(long timestamp, int sequenceNumber) {
+    checkState(firstReceivedTimestamp == C.TIME_UNSET);
+    firstReceivedTimestamp = timestamp;
+  }
+
+  @Override
+  public void consume(
+      ParsableByteArray data, long timestamp, int sequenceNumber, boolean rtpMarker)
+      throws ParserException {
+    checkStateNotNull(trackOutput);
+
+    int sampleOffset = 0;
+    int numSubFrames = getNumOfSubframesFromMpeg4AudioConfig(payloadFormat.fmtpParameters);
+    for (int subFrame = 0; subFrame <= numSubFrames; subFrame++) {
```

Comment:
do you need to loop `numSubFrames + 1` times?

Reply:
With my understanding when the numSubFrames is zero then each RTP packet has one frame of data, similarly if the numSubFrames is 1 then we will have two frames.  Let me know if you think otherwise. 
We are unable to find any clip which has numSubFrames more than zero so this part of code is not tested. Can you share some clips which have non-zero numSubFrames.

Reply:
can you share the spec that defines this? it doesn't seem that it's in RFC3016, maybe it's in ISO14496-3?

Reply:
It is defined in ISO14496-3.
https://csclub.uwaterloo.ca/~ehashman/ISO14496-3-2009.pdf

Reply:
in which section specifically?

Reply:
You can take a look at "Table 1.41 –Table 1.41 – Syntax of AudioMuxElement()".
From the Sec 1.7.3.2.3 numSubFrames is defined as:
"A data element indicating how many PayloadMux() frames are multiplexed (numSubFrames+1). If more than one PayloadMux() frame is multiplexed, all PayloadMux() share a common StreamMuxConfig().The minimum value is 0 indicating 1 subframe. "